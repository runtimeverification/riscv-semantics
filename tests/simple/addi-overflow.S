.text
.globl _start	
_start:
	lui x1, 0x7FFFF
	addi x1, x1, 2047
	addi x1, x1, 2047
	addi x1, x1, 1
	// x1 contains 2^31 - 1, the largest 32-bit two's complement integer.
	// Setting x2 = x1 + 1 should overflow
	addi x2, x1, 1
.globl _halt
_halt:
	nop
