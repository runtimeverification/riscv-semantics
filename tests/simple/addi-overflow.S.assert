regs: {1: 2147483647, 2: -2147483648}
