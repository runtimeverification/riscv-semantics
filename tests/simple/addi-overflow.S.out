<generatedTop>
  <riscv>
    <instrs>
      .K
    </instrs>
    <mem>
      65536 |-> 127 65537 |-> 69 65538 |-> 76 65539 |-> 70 65540 |-> 1 65541 |-> 1 65542 |-> 1 65543 |-> 0 65544 |-> 0 65545 |-> 0 65546 |-> 0 65547 |-> 0 65548 |-> 0 65549 |-> 0 65550 |-> 0 65551 |-> 0 65552 |-> 2 65553 |-> 0 65554 |-> 243 65555 |-> 0 65556 |-> 1 65557 |-> 0 65558 |-> 0 65559 |-> 0 65560 |-> 84 65561 |-> 0 65562 |-> 1 65563 |-> 0 65564 |-> 52 65565 |-> 0 65566 |-> 0 65567 |-> 0 65568 |-> 208 65569 |-> 1 65570 |-> 0 65571 |-> 0 65572 |-> 8 65573 |-> 0 65574 |-> 0 65575 |-> 0 65576 |-> 52 65577 |-> 0 65578 |-> 32 65579 |-> 0 65580 |-> 1 65581 |-> 0 65582 |-> 40 65583 |-> 0 65584 |-> 5 65585 |-> 0 65586 |-> 4 65587 |-> 0 65588 |-> 1 65589 |-> 0 65590 |-> 0 65591 |-> 0 65592 |-> 0 65593 |-> 0 65594 |-> 0 65595 |-> 0 65596 |-> 0 65597 |-> 0 65598 |-> 1 65599 |-> 0 65600 |-> 0 65601 |-> 0 65602 |-> 1 65603 |-> 0 65604 |-> 108 65605 |-> 0 65606 |-> 0 65607 |-> 0 65608 |-> 108 65609 |-> 0 65610 |-> 0 65611 |-> 0 65612 |-> 5 65613 |-> 0 65614 |-> 0 65615 |-> 0 65616 |-> 0 65617 |-> 16 65618 |-> 0 65619 |-> 0 65620 |-> 183 65621 |-> 240 65622 |-> 255 65623 |-> 127 65624 |-> 147 65625 |-> 128 65626 |-> 240 65627 |-> 127 65628 |-> 147 65629 |-> 128 65630 |-> 240 65631 |-> 127 65632 |-> 147 65633 |-> 128 65634 |-> 16 65635 |-> 0 65636 |-> 19 65637 |-> 129 65638 |-> 16 65639 |-> 0 65640 |-> 19 65641 |-> 0 65642 |-> 0 65643 |-> 0
    </mem>
    <regs>
      1 |-> 2147483647 2 |-> -2147483648
    </regs>
    <pc>
      65640
    </pc>
  </riscv>
  <test>
    <halt>
      ADDRESS ( 65640 )
    </halt>
  </test>
  <generatedCounter>
    0
  </generatedCounter>
</generatedTop>